src/ff/curve_points.rs

```diff
@@ -0,0 +1,272 @@
+    }
+}
+
+impl std::ops::Neg for RP25519 {
+    type Output = Self;
+
+    fn neg(self) -> Self::Output {
+        Self(self.0.decompress().unwrap().neg().compress())
+    }
+}
+
+impl std::ops::Sub for RP25519 {
+    type Output = Self;
+
+    fn sub(self, rhs: Self) -> Self::Output {
+        Self((self.0.decompress().unwrap() - rhs.0.decompress().unwrap()).compress())
+    }
+}
+
+impl std::ops::SubAssign for RP25519 {
+    #[allow(clippy::assign_op_pattern)]
+    fn sub_assign(&mut self, rhs: Self) {
+        *self = *self - rhs;
+    }
+}
+
+///Scalar Multiplication
+///<'a, 'b> `std::ops::Mul<&'b"` Fp25519 for &'a
+impl RP25519 {
+
+    /// # Errors
+    /// Propagates errors from decompressing invalid curve point
+    pub fn s_mul(self, rhs: Fp25519) -> Result<RP25519, Error> {
+        self.0
+            .decompress()
+            .map_or(Err(Error::DecompressingInvalidCurvePoint), |x| {
+                Ok((x * Scalar::from(rhs)).compress().into())
+            })
+    }
+}
+
+///do not use
+impl std::ops::Mul for RP25519 {
+    type Output = Self;
+
+    fn mul(self, _rhs: RP25519) -> Self::Output {
+        panic!("Two curve points cannot be multiplied! Do not use *, *= for RP25519 or secret shares of RP25519");
+    }
+}
+
+///do not use
+impl std::ops::MulAssign for RP25519 {
+    fn mul_assign(&mut self, _rhs: RP25519) {
+        panic!("Two curve points cannot be multiplied! Do not use *, *= for RP25519 or secret shares of RP25519");
+    }
+}
+
+impl From<Scalar> for RP25519 {
+    fn from(s: Scalar) -> Self {
+        RP25519(RistrettoPoint::mul_base(&s).compress())
+    }
+}
+
+impl From<Fp25519> for RP25519 {
+    fn from(s: Fp25519) -> Self {
+        RP25519(RistrettoPoint::mul_base(&s.into()).compress())
+    }
+}
+
+impl From<CompressedRistretto> for RP25519 {
+    fn from(s: CompressedRistretto) -> Self {
+        RP25519(s)
+    }
+}
+
+impl From<RP25519> for CompressedRistretto {
+    fn from(s: RP25519) -> Self {
+        s.0
+    }
+}
+
+macro_rules! cp_hash_impl {
+    ( $u_type:ty, $byte_size:literal) => {
+        impl From<RP25519> for $u_type {
+            fn from(s: RP25519) -> Self {
+                let hk = Hkdf::<Sha256>::new(None, s.0.as_bytes());
+                let mut okm = [0u8; $byte_size];
+                //error invalid length from expand only happens when okm is very large
+                hk.expand(&[], &mut okm).unwrap();
+                <$u_type>::from_le_bytes(okm)
+            }
+        }
+
+        impl From<$u_type> for RP25519 {
+            fn from(s: $u_type) -> Self {
+                let hk = Hkdf::<Sha256>::new(None, &s.to_le_bytes());
+                let mut okm = [0u8; 32];
+                //error invalid length from expand only happens when okm is very large
+                hk.expand(&[], &mut okm).unwrap();
+                RP25519::deserialize(&okm.into())
+            }
+        }
+    };
+}
+
+cp_hash_impl!(u64, 8);
+
+cp_hash_impl!(u32, 4);
```

Comment:
do you want to use it in tests only?

Reply:
need them to convert curve points into pseudonyms, i.e. u64 (maybe u128 is better since there are less collisions)

Reply:
I am asking because I saw usages of `u64` but `u32` seems to be only used in tests. If that's true, it is better to move this implementation behind `test` feature